### PR TITLE
Russell-LB123.Electricity

### PR DESCRIPTION
**Pull request by russellhz**
#681

### CODE REVIEW

File: R/zchunk_LB123.Electricity.R [hunk 1]
```diff
@@ -1,20 +1,19 @@
 #' module_gcam.usa_LB123.Electricity
 #'
-#' Briefly describe what this chunk does.
+#' By state, calculates electricity fuel consumption, electricity generation, and inputs and outputs of net ownuse.
```
Comment (bpbond) [inline R/zchunk_LB123.Electricity.R]: "ownuse"?

File: inst/extdata/gcam-usa/EIA_elect_td_ownuse_prices.csv [hunk 2]
```diff
@@ -0,0 +1,58 @@
+# File: EIA_elect_td_ownuse_prices.csv
+# Title: EIA State Electricity Profiles 2008
+# Description: Table A1. Selected Electric Industry Summary Statistics by State
+# Source: ftp://eia.gov/electricity/stateprofiles/08st_profiles/table_a.pdf
+# Units: In column names
```
Comment (bpbond) [inline inst/extdata/gcam-usa/EIA_elect_td_ownuse_prices.csv]: What is `c.kWh` is Price column?

Comment (russellhz) [reply inst/extdata/gcam-usa/EIA_elect_td_ownuse_prices.csv]: I believe it's cents/kWh (maybe in 2008$)

File: R/zchunk_LB123.Electricity.R (hunk 1 above)
Comment (russellhz) [reply R/zchunk_LB123.Electricity.R]: ownuse is the electricity consumed by the power plants or transformation process.

File: R/zchunk_LB123.Electricity.R [hunk 3]
```diff
@@ -34,82 +33,175 @@ module_gcam.usa_LB123.Electricity_DISABLED <- function(command, ...) {
...
+      repeat_add_columns(tibble(year = HISTORICAL_YEARS)) %>%
+      # Add in ownuse by state
+      left_join_error_no_match(EIA_elect_td_ownuse_prices %>%
+                                 select(State, DirectUse_MWh), by = c("state" = "State")) %>%
+      group_by(sector, fuel, year) %>%
+      # Compute state share of total
+      mutate(value = DirectUse_MWh / sum(DirectUse_MWh)) %>%
+      ungroup()
+
+    # Net own use = national total by each state's share
```
Comment (gokuliyer) [inline R/zchunk_LB123.Electricity.R]: Should this be "Net own use = national total multiplies by each state's share"?

File: R/zchunk_LB123.Electricity.R [hunk 4]
```diff
@@ -34,82 +33,175 @@ module_gcam.usa_LB123.Electricity_DISABLED <- function(command, ...) {
...
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
+    # SEDS (EIA) indicates electricity generation technologies either in terms of fuel inputs or fuel outputs (not both)
+    # ELECTRICITY_INPUT: coal, gas, oil, biomass
+    # ELECTRICITY_OUTPUT: nuclear and renewables
+    L123.pct_state_elec_F <- L101.inEIA_EJ_state_S_F %>%
+      filter(sector %in% c("electricity_input", "electricity_output")) %>%
+      # Compute each state's percentage, by fuel
+      group_by(sector, fuel, year) %>%
+      mutate(value = value / sum(value)) %>%
```
[the remark this answers is not in the document]
Comment (gokuliyer) [reply R/zchunk_LB123.Electricity.R]: I see that it's just the energy inputs by state. Never mind the above comment.

File: R/zchunk_LB123.Electricity.R [hunk 5]
```diff
@@ -34,82 +33,175 @@ module_gcam.usa_LB123.Electricity_DISABLED <- function(command, ...) {
...
+    # Electricity generation inputs by fuel and state
+    # Allocating total energy input values to states using shares
+    L123.in_EJ_state_elec_F <- L123.pct_state_elec_F %>%
+      # L123.in_EJ_R_elec_F_Yh only has certain fuels
+      filter(fuel %in% unique(L123.in_EJ_R_elec_F_Yh$fuel)) %>%
+      left_join_error_no_match(L123.in_EJ_R_elec_F_Yh %>%
+                                 select(fuel, year, value),
+                               by = c("fuel", "year")) %>%
+      # Multiplying state share by total value
+      mutate(value = value.x * value.y,
```
[the remark this answers is not in the document]
Comment (gokuliyer) [reply R/zchunk_LB123.Electricity.R]: Again, don't mind the above.

**Comment by bpbond** [comment]
Waiting on @gokuliyer approval.

**Comment by gokuliyer** [comment]
I approve.

**Pull request merged**